webapp/lib/view.dart

```diff
@@ -937,127 +937,267 @@ class ResponseListView extends BaseView {
   }
 }
 
-enum FormGroupTypes {
-  TEXT,
-  CHECKBOX,
-  DATE
-}
-
 class ProjectConfigurationView extends BaseView{
-  DivElement configurationViewElement;
+  DivElement _configurationViewElement;
   FormElement _projectConfigurationForm;
+  Map _formData;
+  List<String> additionalProjectLanguages;
 
-  ProjectConfigurationView() {
-    configurationViewElement = new DivElement()
+  ProjectConfigurationView(Map formData, this.additionalProjectLanguages) {
+    _formData = new Map.from(formData);
```

Comment:
unfortunately `Map.from()` doesn't make a deep copy, only a shallow one. If the map only had primitive types (nums, strings, bools) that would be fine, but this map also contains lists and other maps, so the new Map will maintain a reference to those rather than make a copy. This means that any changes to those will be reflected in the original map as well.

It's not a very nice solution, but since we only have maps and lists, we can turn the data into json and back to make a deep copy of it:
```suggestion
    _formData = json.decode(json.encode(formData));
```
We'll likely want to move to objects encapsulating this data anyway, so we'll be able to implement a copy method more elegantly then.

Reply:
Good to know about that. I've made the change